app/celery/tasks.py

```diff
@@ -47,16 +47,20 @@ def zip_and_send_letter_pdfs(filenames_to_zip, upload_filename):
         )
     except ClientError:
         current_app.logger.exception('FTP app failed to download PDF from S3 bucket {}'.format(folder_date))
-        task_name = "update-letter-notifications-to-error"
+        self.retry(queue='process-ftp-tasks')
+        return
     except FtpException:
         try:
             # check if file exists with the right size.
             # It has happened that an IOError occurs but the files are present on the remote server.
             ftp_client.file_exists_with_correct_size(upload_filename, len(zip_data))
             task_name = "update-letter-notifications-to-sent"
         except FtpException:
-            current_app.logger.exception('FTP app failed to send api messages')
-            task_name = "update-letter-notifications-to-error"
+            current_app.logger.exception('Max retry failed: FTP app failed to send api messages')
```

Comment:
```suggestion
            current_app.logger.exception(f'Failed to send {} to FTP server. Retrying.'.format(zips_sent_filename))
```

Is the original log meant to go in the handler below? In both cases, I think it'll be helpful if we include some identifying info in the log.

Reply:
We could also append the exception message, which also contains [some useful data](https://github.com/alphagov/notifications-ftp/blob/c57c0a0d910b3f0f54acb1a91de59d8f987b090f/app/sftp/ftp_client.py#L88).